src/components/laser-controls.js

```diff
@@ -1,69 +1,79 @@
 
 registerComponent('laser-controls', {
   schema: {
-    hand: {default: 'right'}
+    hand: {default: 'right'},
+    model: {default: true}
   },
 
   init: function () {
-    var config = this.config;
     var data = this.data;
     var el = this.el;
-    var self = this;
+    var controllerConfig = {hand: data.hand, model: data.model};
 
     // Set all controller models.
-    el.setAttribute('daydream-controls', {hand: data.hand});
-    el.setAttribute('gearvr-controls', {hand: data.hand});
-    el.setAttribute('oculus-touch-controls', {hand: data.hand});
-    el.setAttribute('vive-controls', {hand: data.hand});
-    el.setAttribute('windows-motion-controls', {hand: data.hand});
+    el.setAttribute('daydream-controls', controllerConfig);
+    el.setAttribute('gearvr-controls', controllerConfig);
+    el.setAttribute('oculus-touch-controls', controllerConfig);
+    el.setAttribute('vive-controls', controllerConfig);
+    el.setAttribute('windows-motion-controls', controllerConfig);
 
-    // Wait for controller to connect, or have a valid pointing pose, before creating ray
-    el.addEventListener('controllerconnected', createRay);
-    el.addEventListener('controllerdisconnected', hideRay);
-    el.addEventListener('controllermodelready', function (evt) {
-      createRay(evt);
-      self.modelReady = true;
-    });
-
-    function createRay (evt) {
-      var controllerConfig = config[evt.detail.name];
-
-      if (!controllerConfig) { return; }
-
-      // Show the line unless a particular config opts to hide it, until a controllermodelready
-      // event comes through.
-      var raycasterConfig = utils.extend({
-        showLine: true
-      }, controllerConfig.raycaster || {});
+    // Bind methods
+    this.createRay = bind(this.createRay, this);
+    this.hideRay = bind(this.hideRay, this);
+    this.setModelReady = bind(this.setModelReady, this);
 
-      // The controllermodelready event contains a rayOrigin that takes into account
-      // offsets specific to the loaded model.
-      if (evt.detail.rayOrigin) {
-        raycasterConfig.origin = evt.detail.rayOrigin.origin;
-        raycasterConfig.direction = evt.detail.rayOrigin.direction;
-        raycasterConfig.showLine = true;
-      }
-
-      // Only apply a default raycaster if it does not yet exist. This prevents it overwriting
-      // config applied from a controllermodelready event.
-      if (evt.detail.rayOrigin || !self.modelReady) {
-        el.setAttribute('raycaster', raycasterConfig);
-      } else {
-        el.setAttribute('raycaster', 'showLine', true);
-      }
-
-      el.setAttribute('cursor', utils.extend({
-        fuse: false
-      }, controllerConfig.cursor));
+    // Wait for controller to connect, or have a valid pointing pose, before creating ray
+    el.addEventListener('controllerconnected', this.createRay);
+    el.addEventListener('controllerdisconnected', this.hideRay);
+    el.addEventListener('controllermodelready', this.createRay);
```

Comment:
Why two separate event listeners attached to the same event?

Reply:
Fixed in 59d2430